scrapy/item.py

```diff
@@ -14,28 +14,39 @@
 from scrapy.utils.trackref import object_ref
 
 
-class BaseItem(object_ref):
-    """Base class for all scraped items.
+class _BaseItem(object_ref):
+    """
+    Temporary class used internally to avoid the deprecation
+    warning raised by isinstance checks using BaseItem.
+    """
+    pass
 
-    In Scrapy, an object is considered an *item* if it is an instance of either
-    :class:`BaseItem` or :class:`dict`. For example, when the output of a
-    spider callback is evaluated, only instances of :class:`BaseItem` or
-    :class:`dict` are passed to :ref:`item pipelines <topics-item-pipeline>`.
 
-    If you need instances of a custom class to be considered items by Scrapy,
-    you must inherit from either :class:`BaseItem` or :class:`dict`.
+class _BaseItemMeta(ABCMeta):
```

Comment:
It looks like we are trying to reinvent `create_deprecated_class` util from https://github.com/scrapy/scrapy/blob/794ab19660d369f273abdd5b93721c209f6e4eab/scrapy/utils/deprecate.py#L18-L121

Reply:
I looked (briefly) at that function before doing these changes, for some reason it didn't appear appropriate for this case. But I can't actually remember what that reason was, so I'll check again 👍

Reply:
Right, I tried this and remembered why I didn't use it. If I understand correctly, what you suggest is to do `BaseItem = create_deprecated_class('BaseItem', Item)`. The problem I see is that we don't want the deprecated `BaseItem` to have the functionality of `Item`, i.e. the field capabilities provided by the `ItemMeta` meta class:

```python
In [1]: from scrapy.item import Item, Field 
   ...: from scrapy.utils.deprecate import create_deprecated_class 
   ...:  
   ...: BaseItem = create_deprecated_class('BaseItem', Item, old_class_path="scrapy.item.BaseItem")  
   ...:  
   ...: class TestItem(BaseItem): 
   ...:     foo = Field() 
   ...:  
   ...: print(TestItem(foo="bar"))                                                                                                                                                                                                            
/.../scrapy/venv-scrapy/bin/ipython:6: ScrapyDeprecationWarning: __main__.TestItem inherits from deprecated class scrapy.item.BaseItem, please inherit from scrapy.item.Item. (warning only on first subclass, there may be others)
  
{'foo': 'bar'}
```

@dangra Please correct me if I'm wrong. Thanks!

Reply:
Ping @dangra 😄